blocks/main_loop.py

```diff
@@ -168,7 +168,7 @@ def run(self):
                         extension.main_loop = self
                     self._run_extensions('before_training')
                     with Timer('initialization', self.profile):
-                        self.algorithm.initialize()
+                        self.algorithm.initialize(profile=self.profile)
```

Comment:
No need for this I don't think --- we have two different notions of "profiling", one is the Theano profiler which profiles calls to compiled Theano functions, and one that is the Blocks-level profiler.

The question is how you would pass **kwargs to `theano.function` via a call to `run()`. We could just add `**kwargs` to `run()` and pass them along to `initialize()` on this line, but... I don't know. 

@rizar curious to hear your opinion of how this should work, you usually have good design ideas about things like this.